scripts/deploy.py

```diff
@@ -40,22 +40,36 @@ def cli(ctx, start, list, steps):
 def new_minor_release(ctx):
     p = ctx.obj['pipeline']
 
+    # version.inc update strategy:
+    # 1. Merge everything from master to release (version file is not changed
+    #    because it was already merged at step 3 of previous iteration)
+    # 2. Change version.inc in release branch to new "dev"
+    # 3. Merge release to master (so that master has new "dev" too)
+    # 4. Change version.inc in release branch to new release version.
+
+    # Then do the push, so all that merging back and forth is kind of atomic
+    # from github point of view).
+
+    # Without that (e.g. modifying version.inc independently rather than having
+    # a sequence), all merges will have a collision in version.inc and that's
+    # painful to resolve.
```

Comment:
Er, see my other comment that I wrote 30 minutes ago as a reply to previous comments, but perhaps I misunderstood you. Perhaps by "collision" you actually mean "git conflict"? At any rate, I still don't quite see how doing this the simple way would generate merge conflicts. Suffice it to say, I really think doing any release->master merges, ever in the history of ever, is a bad idea.

Reply:
Oh, I suppose that having patch versions on release separately from master might generate conflicts. Even so, I'd rather build this script to handle conflicts than do any sort of backwards merges.